src/charm.py

```diff
@@ -3,115 +3,90 @@
-        environment configuration for your specific workload.
+class ContentCacheBackendsConfigCharm(ops.CharmBase):
+    """Charm the application."""
 
-        Learn more about interacting with Pebble at at https://juju.is/docs/sdk/pebble.
+    def __init__(self, framework: ops.Framework) -> None:
+        """Initialize the object.
 
         Args:
-            event: event triggering the handler.
+            framework: The ops framework.
         """
-        # Get a reference the container attribute on the PebbleReadyEvent
-        container = event.workload
-        # Add initial Pebble config layer using the Pebble API
-        container.add_layer("httpbin", self._pebble_layer, combine=True)
-        # Make Pebble reevaluate its plan, ensuring any services are started if enabled.
-        container.replan()
-        # Learn more about statuses in the SDK docs:
-        # https://juju.is/docs/sdk/constructs#heading--statuses
+        super().__init__(framework)
+        framework.observe(self.on.start, self._on_start)
+        framework.observe(self.on.config_changed, self._on_config_changed)
+        framework.observe(
+            self.on[CACHE_CONFIG_INTEGRATION_NAME].relation_changed,
+            self._on_cache_config_relation_changed,
+        )
+        framework.observe(
+            self.on[CACHE_CONFIG_INTEGRATION_NAME].relation_broken,
+            self._on_cache_config_relation_broken,
+        )
+
+    def _on_start(self, _: ops.StartEvent) -> None:
+        """Handle start event."""
+        self._set_status()
+
+    def _on_config_changed(self, _: ops.ConfigChangedEvent) -> None:
+        """Handle config changed event."""
+        self._load_integration_data()
+
+    def _on_cache_config_relation_changed(self, _: ops.RelationChangedEvent) -> None:
+        """Handle cache config relation changed event."""
+        self._load_integration_data()
+
+    def _on_cache_config_relation_broken(self, _: ops.RelationBrokenEvent) -> None:
+        """Handle cache config relation broken event."""
+        self._set_status()
+
+    def _load_integration_data(self) -> None:
+        """Validate the configuration and load to integration."""
+        if not self.unit.is_leader():
+            logger.debug("Not leader: not setting the integration data")
+            return
+
+        logger.info("Loading configuration")
+        try:
+            config = Configuration.from_charm(self)
+        except ConfigurationError as err:
+            logger.error("Configuration error: %s", err)
+            self.unit.status = ops.BlockedStatus(str(err))
+            return
+
+        logger.info("Setting integration data")
+        if self.model.relations[CACHE_CONFIG_INTEGRATION_NAME]:
+            rel = self.model.relations[CACHE_CONFIG_INTEGRATION_NAME][0]
+            rel.data[self.app].update(config.to_integration_data())
+        logger.info("Integration data set")
+        self._set_status()
```

Comment:
This method duplicates some code already executed in this method, I think some factoring can be done.
Call the "_set_status" at the beginning of "_load_integration_data"
After the call if the status is not Active, return immediately
That will allow you to remove the duplicate of "is_leader" and the check of relation presence in l72

Reply:
Updated.